client/state/selectors/can-site-view-atomic-hosting.js

```diff
@@ -29,13 +30,21 @@ export default function canSiteViewAtomicHosting( state ) {
 	// This is also enforced on the server, please remove client checks later.
 	// ID of site added 31 Oct 2019, so only sites newer currently eligible
 	if ( siteId > 168768859 ) {
+		const canManageOptions = canCurrentUser( state, siteId, 'manage_options' );
+
+		if ( ! canManageOptions ) {
+			return false;
+		}
+
 		const isAtomicSite = !! isSiteAutomatedTransfer( state, siteId );
+		const planSlug = get( getSelectedSite( state ), [ 'plan', 'product_slug' ] );
+		const hasEligablePlan = isBusinessPlan( planSlug ) || isEcommercePlan( planSlug );
 
-		if ( isAtomicSite ) {
+		if ( isAtomicSite && hasEligablePlan ) {
 			return true;
 		}
 
-		if ( isEnabled( 'hosting/non-atomic-support' ) ) {
+		if ( ! isAtomicSite && hasEligablePlan && isEnabled( 'hosting/non-atomic-support' ) ) {
 			return isBusinessPlan( get( getSelectedSite( state ), [ 'plan', 'product_slug' ] ) );
 		}
```

Comment:
I think we can simplify this to just return `return hasEligiblePlan && isEnabled( 'hosting/non-atomic-support' );`.